torchtext/models/roberta/modules.py

```diff
@@ -124,6 +124,10 @@ def forward(self, query, key_padding_mask):
         )
 
         attn_weights = torch.bmm(q, k.transpose(1, 2))
+        if attn_mask is not None:
+            torch._assert(attn_mask.dim() == 2, "Expected attn_mask of dim 2 but got {}".format(attn_mask.dim()))
```

Comment:
At this point I wonder if it makes sense to explicitly check that the sizes of attn_mask and attn_weights match exactly (or right after attn_mask.unsqueeze)

Reply:
yupp make sense. Added the checks!